scripts/vlsi_rom_gen

```diff
@@ -81,22 +76,16 @@ def iterate_by_n(it, n):
         ...
     ValueError: Iterable length not evenly divisible by 4
     """
-    it = iter(it)
-    while True:
+    if len(li) % n != 0:
+        raise ValueError(
+            'Iterable length not evenly divisible by {}'.format(n)
+        )
+    for i in range(len(li) // n):
         batch = ()
-        for i in range(n):
```

Comment:
I don't have time to review this thoroughly. Is changing this code around necessary?

Reply:
Yes. The generator behavior changed since python3.7. You could refer to the stackoverflow link for more details.